code/game/objects/items/weapons/landmines.dm

```diff
@@ -0,0 +1,146 @@
+/obj/item/weapon/landmine
+	name = "land mine"
+	desc = "An anti-personnel explosive device used for area denial."
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "landmine"
+	throwforce = 0
+	w_class = 3
+	var/deployed = 0
+
+/obj/item/weapon/landmine/update_icon()
+	..()
+
+	if(!deployed)
+		icon_state = "[icon_state]"
+	else
+		icon_state = "[icon_state]_on"
+
+/obj/item/weapon/landmine/verb/hide_under()
+	set src in oview(1)
+	set name = "Hide"
+	set category = "Object"
+
+	if(use_check(user, USE_DISALLOW_SILICONS))
+		return
+
+	layer = TURF_LAYER+0.2
+	usr << "<span class='notice'>You hide \the [src].</span>"
+
+
+/obj/item/weapon/landmine/attack_self(mob/user as mob)
+	..()
+	if(!deployed && !use_check(user, USE_DISALLOW_SILICONS))
+		user.visible_message(
+			"<span class='danger'>[user] starts to deploy \the [src].</span>",
+			"<span class='danger'>You begin deploying \the [src]!</span>"
+			)
+
+		if (do_after(user, 60))
+			user.visible_message(
+				"<span class='danger'>[user] has deployed \the [src].</span>",
+				"<span class='danger'>You have deployed \the [src]!</span>"
+				)
+
+			deployed = 1
+			user.drop_from_inventory(src)
+			update_icon()
+			anchored = 1
+
+/obj/item/weapon/landmine/proc/trigger(mob/living/L)
+	spark(src, 3, alldirs)
+	if(L)
+		if(ishuman(L))
+			L.Weaken(2)
+	explosion(loc, 0, 1, 2, 3)
+	qdel(src)
+
+/obj/item/weapon/landmine/Crossed(AM as mob|obj)
+	if(deployed && isliving(AM))
+		var/mob/living/L = AM
+		if(L.mob_size >= 5)
+			L.visible_message(
```

Comment:
Maybe add a click sound for blind people (3rd param)?

Reply:
Fixed.